tlsf/CMakeLists.txt

```diff
@@ -7,6 +7,14 @@ if(WIN32 OR APPLE OR ANDROID)
   return()
 endif()
 
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
```

Comment:
So the thing about this is that this package is completely C.  So I think we should instead do:

```
# Default to C11
if(NOT CMAKE_C_STANDARD)
  set(CMAKE_C_STANDARD 11)
endif()
```

We should also fix up the `CMAKE_COMPILER_IS_GNUCXX` to be something like `CMAKE_COMPILER_IS_GNUC`; while the former will always work with our system, the latter is more "correct" (note that I just made up that variable name; you'll have to go searching for what it is actually called).

Reply:
I believe that [`CMAKE_COMPILER_IS_GNUCC`](https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILER_IS_GNUCC.html?highlight=cmake_c_compiler_id) and [`CMAKE_C_COMPILER_ID`](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html#variable:CMAKE_%3CLANG%3E_COMPILER_ID) are the variables that I should use. My most recent commit reflects this. Although, your guess, `CMAKE_COMPILER_IS_GNUC` (I was able to see that a warning was thrown if I undid a change to fix a warning).